server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraMessageIdManagerInjectionTest.java

```diff
@@ -23,56 +23,31 @@
 
 import javax.inject.Inject;
 
-import org.apache.commons.configuration.ConfigurationException;
-import org.apache.commons.configuration.HierarchicalConfiguration;
 import org.apache.james.lifecycle.api.Configurable;
 import org.apache.james.mailbox.MessageIdManager;
 import org.apache.james.utils.ConfigurationPerformer;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.ClassRule;
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
 import com.google.common.collect.ImmutableList;
-import com.google.inject.AbstractModule;
-import com.google.inject.Module;
 import com.google.inject.multibindings.Multibinder;
 
-public class CassandraMessageIdManagerInjectionTest {
-
-    @ClassRule
-    public static DockerCassandraRule cassandra = new DockerCassandraRule();
-    
-    @Rule
-    public CassandraJmapTestRule cassandraJmap = CassandraJmapTestRule.defaultTestRule();
-    
-    private GuiceJamesServer server;
-
-    @Before
-    public void setup() throws Exception {
-        Module module = new AbstractModule() {
-            @Override
-            protected void configure() {
-                Multibinder.newSetBinder(binder(), ConfigurationPerformer.class).addBinding().to(CallMe.class);
-            }
-        };
-        server = cassandraJmap.jmapServer(module, cassandra.getModule());
-        server.start();
-    }
+class CassandraMessageIdManagerInjectionTest {
+    @RegisterExtension
+    static JamesServerExtension testExtension = new JamesServerExtension(
+        CassandraJamesDefinition.builder()
+            .defaultExtensions()
+            .addModules(binder -> Multibinder.newSetBinder(binder, ConfigurationPerformer.class)
+                .addBinding()
+                .to(CallMe.class))
+            .build());
 
     @Test
-    public void messageIdManagerShouldBeInjected() {
+    void messageIdManagerShouldBeInjected() {
```

Comment:
What is the purpose of this test since it is empty?

Reply:
We are testing that the server starts correctly.

I can be done in a more readable way by:
 - Passing `autoStart = false` to the extension
 - and then manually calling `start` in the method body.

Reply:
https://github.com/linagora/james-project/pull/1763/commits/b649d0cc25e0a36ba712ce136b9fbb5d0378af0c propose a rewrite of this tests, that should be more readable.

Reply:
Thanks